certs/certs.go

```diff
@@ -141,29 +141,28 @@ func ValidateFinalityCertificates(verifier gpbft.Verifier, network gpbft.Network
 // any other parts of the certificate, just that the _value_ has been signed by a majority of the
 // power.
 func verifyFinalityCertificateSignature(verifier gpbft.Verifier, powerTable gpbft.PowerEntries, nn gpbft.NetworkName, cert FinalityCertificate) error {
-	totalPower := new(gpbft.StoragePower)
-	for i := range powerTable {
-		totalPower = totalPower.Add(totalPower, powerTable[i].Power)
+	scaled, totalScaled, err := powerTable.Scaled()
+	if err != nil {
+		return xerrors.Errorf("failed to scale power table: %w", err)
 	}
 
 	signers := make([]gpbft.PubKey, 0, len(powerTable))
-	signerPowers := new(gpbft.StoragePower)
+	var signerPowers uint16
 	if err := cert.Signers.ForEach(func(i uint64) error {
 		if i >= uint64(len(powerTable)) {
 			return xerrors.Errorf(
 				"finality certificate for instance %d specifies a signer %d but we only have %d entries in the power table",
 				cert.GPBFTInstance, i, len(powerTable))
 		}
-		signer := &powerTable[i]
-		signerPowers = signerPowers.Add(signerPowers, signer.Power)
-		signers = append(signers, signer.PubKey)
+		signerPowers += scaled[i]
```

Comment:
Now that we're rounding, SPs can have zero power (rounded down). Reject them here I assume?

Reply:
This touches https://github.com/filecoin-project/go-f3/issues/319 as well. SP's voting eligibility status can change.